common/src/main/java/software/tnb/common/openshift/OpenshiftClient.java

```diff
@@ -628,6 +628,38 @@ public void createDeployment(Map<String, Object> config) {
         });
     }
 
+    public void createDeployment(Map<String, Object> config, software.tnb.common.deployment.WithVolumes volumeProvider) {
+        createDeployment(config, volumeProvider, builder -> {
+        });
+    }
+
+    public void createDeployment(Map<String, Object> config, software.tnb.common.deployment.WithVolumes volumeProvider,
+        Consumer<DeploymentBuilder> customization) {
+        // Add volumes and volumeMounts from the WithVolumes interface to the config
+        java.util.Map<String, Object> enrichedConfig = new java.util.HashMap<>(config);
+
+        List<Volume> volumes = volumeProvider.volumes();
+        List<VolumeMount> volumeMounts = volumeProvider.volumeMounts();
+
+        if (volumes != null && !volumes.isEmpty()) {
+            enrichedConfig.merge("volumes", volumes, (existing, newVal) -> {
+                List<Volume> combined = new java.util.ArrayList<>((List<Volume>) existing);
+                combined.addAll((List<Volume>) newVal);
+                return combined;
+            });
+        }
+
+        if (volumeMounts != null && !volumeMounts.isEmpty()) {
+            enrichedConfig.merge("volumeMounts", volumeMounts, (existing, newVal) -> {
+                List<VolumeMount> combined = new java.util.ArrayList<>((List<VolumeMount>) existing);
+                combined.addAll((List<VolumeMount>) newVal);
+                return combined;
+            });
+        }
+
+        createDeployment(enrichedConfig, customization);
+    }
+
     /**
```

Comment:
`config` already handles volumes and volumemounts, so just add them into the config map in the services, instead of adding extra methods here for a specific case